### PR TITLE
refactor: Drop unused `QFrame`s in `SendCoinsEntry`

### PR DESCRIPTION
**Pull request by hebasto**
The `SendCoins_UnauthenticatedPaymentRequest` and `SendCoins_AuthenticatedPaymentRequest` sub-`QFrame`'s of the `SendCoinsEntry` widget have been unused since bitcoin/bitcoin#17165.

Removed all dead code. The resulted `SendCoinsEntry` widget has been simplified.

### CODE REVIEW

**Review by furszy — APPROVED** [review]
Code reviewed ACK a57cd028

No functional changes.

File: src/qt/sendcoinsentry.h [hunk 1]
```diff
@@ -25,7 +25,7 @@ namespace Ui {
  * Stacked widget, with different UIs for payment requests
```
Comment (laanwj) [inline src/qt/sendcoinsentry.h:25]: Please change the comment too

**Review by shaavan — APPROVED** [review]
ACK a57cd028f5354e0fcc872aca6358ef7b1287b191

- This PR does an excellent clean-up to `sendcoinentry.ui` file by:
    - Removing no longer used code.
    - Simplifying the widget type for the updated UI elements.
    - Fixing indentations.
    - Fixing the parent class for the `BitcoinAmountField` widget.

- I was able to successfully compile and run this PR on Ubuntu 22.04 with `Qt version 5.15.3`

As @laanwj rightly pointed out. The comment mentioning “_Stacked widget_” shall also be removed.

**Comment by hebasto** [comment]
Updated a57cd028f5354e0fcc872aca6358ef7b1287b191 -> 7ab72b9d2a500420a1103221c3cbfac4cda1c43a ([pr612.01](https://github.com/hebasto/gui/commits/pr612.01) -> [pr612.02](https://github.com/hebasto/gui/commits/pr612.02), [diff](https://github.com/hebasto/gui/compare/pr612.01..pr612.02)):

- addressed @laanwj's [comment](https://github.com/bitcoin-core/gui/pull/612#discussion_r892534078)

File: src/qt/sendcoinsentry.h (hunk 1 above)
Comment (hebasto) [reply src/qt/sendcoinsentry.h]: @laanwj 

Thank you! [Updated](https://github.com/bitcoin-core/gui/pull/612#issuecomment-1153138515).

**Review by w0xlt — APPROVED** [review]
Tested ACK https://github.com/bitcoin-core/gui/pull/612/commits/7ab72b9d2a500420a1103221c3cbfac4cda1c43a

**Review by shaavan — APPROVED** [review]
reACK 7ab72b9d2a500420a1103221c3cbfac4cda1c43a

Changes since my last review:
- @laanwj's comment is rightly addressed.

I searched for comments mentioning "_Stacked ..._" in `src/qt/sendcoinsentry.*` files, and I found none. This shows that all these comments are appropriately removed.

**Pull request merged**